java-2016/homework-9/IOhomework/src/ReadFile.java

```diff
@@ -0,0 +1,97 @@
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.*;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+
+/**
+ * Created by Turian Ovidiu.
+ * This class is responsible with reading a file and parse the data into an List of Athletes.
+ */
+public class ReadFile {
+
+    private static Logger LOGGER = Logger.getLogger("ReadFile");
+
+    private List<Athlete> athletes = new ArrayList<>();
+
+    public List<Athlete> getAthletes() {
+        return athletes;
+    }
+
+    public ReadFile() {
+    }
+
+    /**
+     * This method will read the given file and store the data.
+     * @param file the file we want to read.
+     */
+    public void readFile(File file) {
+        LOGGER.info("Inside read file method.");
+
+        String line;
+        boolean column = true;
+
+        if (!file.exists()) {
+            throw new IllegalArgumentException("The file do not exist");
+        }
+
+        if (file.length() == 0){
+            throw new IllegalArgumentException("The file do not exist");
+        }
+
+        LOGGER.info("Start reading file.");
+        try (BufferedReader getInfo = new BufferedReader(new java.io.FileReader(file))) {
+
+            while ((line = getInfo.readLine()) != null) {
+                if (!column) {
+                    athletes.add(storeData(line));
+                } else {
+                    column = false;
+                }
+            }
+        } catch (FileNotFoundException e) {
```

Comment:
Exception should be thrown further. There is no point to continue computation if the file is not found. Program should exit after the exception is processed.